### PR TITLE
AppTP: Fix app/tracker exception dev setting

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/1202279501986195/1204239874324815/f

### Description
Fixed app tracker exception rules in dev settings.

### Steps to test this PR

This is only for internal builds. Check task description.

### CODE REVIEW

File: app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt [hunk 1]
```diff
@@ -112,7 +113,7 @@ class ExceptionRulesDebugReceiverRegister @Inject constructor(
         vpnStopReason: VpnStopReason,
     ) {
         // only for debug builds
-        if (appBuildConfig.isDebug) {
+        if (appBuildConfig.isDebug || appBuildConfig.isInternalBuild()) {
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt:116]: This change is needed only if we want to revert the changes when VPN is turned OFF and back ON. This is my understanding based on the message displayed on the banner.

Remove this change if we want to keep the changes across restarts.

File: app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt [hunk 2]
```diff
@@ -87,7 +88,7 @@ class ExceptionRulesDebugReceiverRegister @Inject constructor(
 
     override fun onVpnStarted(coroutineScope: CoroutineScope) {
         // only for debug builds
-        if (appBuildConfig.isDebug) {
+        if (appBuildConfig.isDebug || appBuildConfig.isInternalBuild()) {
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt:91]: This change may have an impact on https://app.asana.com/0/1202279501986195/1203536459696139

File: app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt [hunk 3]
```diff
@@ -86,22 +86,19 @@ class ExceptionRulesDebugReceiverRegister @Inject constructor(
     private val shouldSaveRules = AtomicBoolean(true)
 
     override fun onVpnStarted(coroutineScope: CoroutineScope) {
-        // only for debug builds
-        if (appBuildConfig.isDebug) {
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt:90]: Removed condition as this functionality should be the same in debug and release. This is only used in internal builds. The rest of the code in this function is untouched.

File: app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt [hunk 4]
```diff
@@ -111,17 +108,14 @@ class ExceptionRulesDebugReceiverRegister @Inject constructor(
         coroutineScope: CoroutineScope,
         vpnStopReason: VpnStopReason,
     ) {
-        // only for debug builds
-        if (appBuildConfig.isDebug) {
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/rules/ExceptionRulesDebugReceiver.kt:115]: Removed condition as this functionality should be the same in debug and release. This is only used in internal builds. The rest of the code in this function is untouched.

**Review by nshuba — APPROVED** [review]
LGTM and works as expected, thanks!
